### ISSUE TITLE
Prediction performed on CPU

### ISSUE THREAD

**Issue by liningpan** [cosanlab/py-feat#136]
Hi,

I recently discovered that at least in some cases Py-FEAT is not actually using GPUs for inference. I'm not sure if this is expected and if it will be addressed by #133.

Here is the script I used for profiling. 

```py
from feat import Detector
from torch.profiler import profile, ProfilerActivity

detector = Detector(
    face_model="retinaface",
    landmark_model="mobilefacenet",
    au_model="svm",
    emotion_model="resmasknet",
    facepose_model="img2pose",
)

video_location = "./WolfgangLanger_Pexels.mp4"

with profile(activities=[ProfilerActivity.CPU, ProfilerActivity.CUDA], record_shapes=True, use_cuda=True) as prof:
    video_prediction = detector.detect_video(video_location, skip_frames=24)

print(prof.key_averages().table(sort_by="self_cpu_time_total", row_limit=20))
prof.export_chrome_trace("trace.json")

```

The `trace.json` file produced by the script can be visualized in chrome `chrome://tracing`.

We can see that very little work is actually done on the GPU.
![Full profile](https://user-images.githubusercontent.com/17289190/192042144-4997ca38-6bf6-4ba1-a0dc-b3336dcc4cf1.png)

If we zoom in we can see that `mkldnn` is called by conv2d, I was expecting CUDA to be used.
![zoom in cpu](https://user-images.githubusercontent.com/17289190/192042949-682bba5c-e675-478e-a911-f1395502ede8.png)

A section that is actually running on the GPU.
![zoom in gpu/cudnn](https://user-images.githubusercontent.com/17289190/192042689-70d08124-9acf-414b-8c76-efaa866f2abf.png)

Additional Information:
AWS EC2 g4dn.xlarge (4 vCPU, 16 gb ram, 1 x Tesla T4)
Ubuntu 20.04 LTS
NVIDIA Driver 515.65.01
PyTorch 1.12.1 + CUDA 11.3
py-feat 0.4.0

**Comment by ejolly** [comment]
This should be addressed in 0.5.0 as you can change the device when initializing a `Detector`. Please re-open if there's still a problem, but we're actively working to improve GPU functionality including with m1 macs

**Comment by hamster-with-human-hands** [comment]
Hi, could I ask, how can I change the device? 
```detector = Detector( device='gpu')``` does not work for me
